### DISCUSSION TITLE
misaligned data received by subscriber

### DISCUSSION THREAD

**Discussion by amdsobhy** [eProsima/Fast-DDS#2775] [Q&A]
1- I added an additional 4bytes variable in my idl struct
2- I used fastddsgen to regenerate my dds files.
3- I confirmed the new variable is inside my newly generated dds files.
4- I fill the data structure in the publisher and I print its contents then publish
5- I read the data received from the subscriber and print its contents.

what I am seeing is that the location of the data is shifted 4 bytes forward from what I expect it to be.

This is what the publisher is sending:
height: 720 width: 1280 cv_type: 8   <--- (Correct data)

This is what the subscriber is getting:
height: 1280 width: 8 cv_type: 3560 <--- (Misaligned data)

Each of those variables is 4 bytes so the subscriber data is shifted 4 bytes.

Anyone has clue what could be the issue?

Note: This is using shared memory on a local machine.

I was able to isolate the problem further. I have a header struct inside the main struct and when I remove the header struct the data becomes aligned, when I place the struct the misalignment returns again.

This seems like a bug?

**Comment by MiguelCompany** [comment]
I think this may be a bug on Fast DDS Gen, and is most probably related to #2777
